.devcontainer/devcontainer.json

```diff
@@ -1,19 +1,73 @@
 {
-    "dockerComposeFile": ["../development/docker-compose.yml", "../development/docker-compose.debug.yml"],
+    "name": "Nautobot Dev Container",
+    "dockerComposeFile": [
+        "../development/docker-compose.yml",
+        "../development/docker-compose.postgres.yml",
+        "../development/docker-compose.dev.yml",
+        "../development/docker-compose.override.yml"
+    ],
+    "features": {
+        "ghcr.io/devcontainers/features/docker-outside-of-docker:1": {}
+    },
     "service": "nautobot",
-	"workspaceFolder": "/source/",
-	"extensions": [
-		"ms-python.vscode-pylance",
-		"streetsidesoftware.code-spell-checker",
-		"eamodio.gitlens",
-		"oderwat.indent-rainbow",
-		"ms-python.python",
-		"ms-vsliveshare.vsliveshare"
-	],
-	"settings": {
-		"python.pythonPath": "/usr/local/bin/python",
-		"python.analysis.extraPaths": ["/source"],
-		"python.linting.pylintEnabled": true,
-		"python.linting.enabled": true
-	}
+    "shutdownAction": "stopCompose",
+    "customizations": {
+        "vscode": {
+            "extensions": [
+                "DavidAnson.vscode-markdownlint",
+                "dbaeumer.vscode-eslint",
+                "eamodio.gitlens",
+                "EditorConfig.EditorConfig",
+                "esbenp.prettier-vscode",
+                "GitHub.vscode-pull-request-github",
+                "ms-python.python",
+                "ms-python.vscode-pylance",
+                "ms-python.black-formatter",
+                "ms-python.flake8",
+                "ms-python.isort",
+                "samuelcolvin.jinjahtml",
+                "tamasfe.even-better-toml"
+            ],
+            "settings": {
+                // Global editor settings
+                "editor.codeActionsOnSave": {
+                    "source.fixAll.eslint": true,
+                    "source.fixAll.markdownlint": true
+                },
+                "editor.formatOnPaste": true,
+                "editor.formatOnSave": true,
+                // Python settings
+                "[python]": {
+                    "editor.codeActionsOnSave": {
+                        // "source.organizeImports": "explicit"
+                    },
+                    // When rebuilding the devcontainer, `Developer: Reload Window` must be
+                    // executed for the Black formatter to be recognized by the settings.
+                    // See: https://github.com/microsoft/vscode/issues/189839
+                    "editor.defaultFormatter": "ms-python.black-formatter"
```

Comment:
We actually just changed the formatter to use the ruff extension `charliermarsh.ruff`. Since I'm already testing this workflow for the review I can make any necessary changes to get the new formatter working unless you'd like to do that?

Reply:
Great! I really like `ruff`, I'll do the change.

Reply:
Thanks! While you're at it you should merge in the latest changes from upstream. There's a fix in the docker file that's needed after Debian made some changes last week

Reply:
Did you also replaced `flake8` and `isort` with `ruff`?

Reply:
We did. Sorry I forgot to mention that earlier

Reply:
#5055 is still open to replace isort but the plan is to merge that soon, perhaps tomorrow

Reply:
Ok, I pushed the changes. Let me know if it's ok with you.

I guess the only question left is what should I do with the `docker-compose.override.yml`? See my comment about providing Nautobot with a superuser.

Reply:
Let me discuss that with @glennmatthews tomorrow. I'm not opposed to adding the superuser environment variables to the development environment by default. 

I'll finish reviewing the changes tomorrow as well to try to get this merged.

Reply:
I have no objection to adding a superuser to the dev environment.

Reply:
Thanks! Now, where do you think it makes more sense to add it? In `dev.env` or in `override.env` / `docker-compose.override.yml`?

Reply:
I'll put the superuser information in `dev.env`. Keep the override for user specific needs...

Reply:
I think `dev.env`. We'll also need to update `nautobot/docs/development/core/docker-compose-advanced-use-cases.md` "Automatically Creating a Superuser" to state that the superuser is now automatically created and that you'll need to set `NAUTOBOT_CREATE_SUPERUSER=false` in `override.env` to disable it. I can help with the doc updates if you'd like.

Reply:
Yeah, my english is not very good, I prefer to let you update the doc. Thanks.

Reply:
I think the changelog is also missing...

Reply:
I can add that as well. We probably need more than one